impr(timer/job/decl): DB script changes, test case add,

subhrajyotim opened the pull request:
Hello Roman,
I have tried to complete 1 as u had mentioned.
I am not too sure, what all test cases I have to cover, so have added only one test case to test the new changes. Please guide. 

Thanks and regards,
Subhro.

romansmirnov commented on engine/src/main/java/org/camunda/bpm/engine/runtime/Job.java:
```diff
@@ -105,4 +105,9 @@
    */
   String getTenantId();
 
+  String getCaseExecutionId();
```
Please provide some javadoc.

romansmirnov commented on engine/src/main/java/org/camunda/bpm/engine/runtime/Job.java:
```diff
@@ -105,4 +105,9 @@
   String getTenantId();
 
+  String getCaseExecutionId();
+  String getCaseInstanceId();
```
Please provide some javadoc.

romansmirnov commented on engine/src/main/java/org/camunda/bpm/engine/runtime/Job.java:
```diff
@@ -105,4 +105,9 @@
 
+  String getCaseExecutionId();
+  String getCaseInstanceId();
+  String getCaseDefinitionId();
```
Please provide some javadoc.

romansmirnov commented on engine/src/main/java/org/camunda/bpm/engine/runtime/Job.java:
```diff
@@ -105,4 +105,9 @@
+  String getCaseExecutionId();
+  String getCaseInstanceId();
+  String getCaseDefinitionId();
+  String getCaseDefinitionKey();
```
Please provide some javadoc.

romansmirnov commented:
Hi Subhro,

It would be great if you could provide some more testcases (ie. `TimerEventListenerHandlerTest`). For example that the type of the `TimerEventListenerJobDeclaration` is set correctly (ie. to `CYCLE`, `DURATION` or `DATE`). A further testcase would be that the correct `jobHandlerType` (ie. `TimerEventListenerJobHandler.TYPE`) is set.

Could you please update this pull request?

Thanks.

Cheers,
Roman

subhrajyotim commented:
Yes Roman. will do the changes as u mention.

subhrajyotim commented:
Hello Roman,
I have added the tests as u advised. However I dont see a getter for TimerDeclarationType in class TimerJobDeclaration. So i have added a getter and it returns String instead of TimerDeclarationType, since it seemed TimerDeclarationType is an impl class and not meant for public API. Please let me know the correct approach, will fix it. Please let me know what other changes I have to make.

Thanks,
Subhro.

romansmirnov commented:
Hi Subhro,

Thanks for your changes. I merged it into the branch [CAM-4450-timer-event-listener](https://github.com/camunda/camunda-bpm-platform/tree/CAM-4450-timer-event-listener). 

So, the next steps would be:
1) A `JobDefinition` must be created, when deploying a case definition containing a timer event listener
2) Implement the timer event listener job that the event `occurred` is triggered when the timer job is executed.
3) Implement the start trigger
4) Implement repeating timer intervals

I would appreciate it, if you could create for each point a pull request.

If you need some assistance, do not hesitate to ask me.

Thanks for your effort!

Cheers,
Roman

subhrajyotim commented:
Hi Roman,
Sorry for responding late, i was stuck with some work, and was unable to
pick this up, last week. Will finish #1 this week.

Thanks a lot,
Subhro.

On Mon, Apr 11, 2016 at 1:29 PM, Roman Smirnov <notifications@github.com>
wrote:

> Hi Subhro,
>
> Thanks for your changes. I merged it into the branch
> CAM-4450-timer-event-listener
> <https://github.com/camunda/camunda-bpm-platform/tree/CAM-4450-timer-event-listener>.
>
>
> So, the next steps would be:
> 1) A JobDefinition must be created, when deploying a case definition
> containing a timer event listener
> 2) Implement the timer event listener job that the event occurred is
> triggered when the timer job is executed.
> 3) Implement the start trigger
> 4) Implement repeating timer intervals
>
> I would appreciate it, if you could create for each point a pull request.
>
> If you need some assistance, do not hesitate to ask me.
>
> Thanks for your effort!
>
> Cheers,
> Roman
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/camunda/camunda-bpm-platform/pull/222#issuecomment-208213738>
>

subhrajyotim commented:
Hello Roman,

Here is what I have to do to accomplish #1
- Change CmmnDeployer.transformDefinitions(..) method to populate
properties map with TimerJobDeclaration List.
- Not sure how to get the job declarations from the CmmnTransformer object,
or the CaseDefnitionEntity object.
- keep a list of timerjobdeclarations similar to taskList in
CaseDefinitionEntity,
- use the TimerEventListenerItemHandler to transfer the declarations to the
CaseDefinitionEntity, this will give the timerdeclaration objects to the
CMMNDeployer. This might not be correct, so please guide.

- Change JobDefinition to have caseInstanceId, caseDefinitionId and
caseDefinitionKey
- Change JobDefinitionEntity to impl the same change
- Change the queries in JobDefinition.xml
- in CmmnDeployer override definitionAddedToDeploymentCache(...) method to
create JobDefinitions from TimerJobDeclarations.
- as part of definitionAddedToDeploymentCache (..) impl insert job
definitions as required, or load existing ones.

Pl help with this understanding. Please guide.

Thanks a lot for your time.
Subhro.


On Mon, Apr 11, 2016 at 1:29 PM, Roman Smirnov <notifications@github.com>
wrote:

> Hi Subhro,
>
> Thanks for your changes. I merged it into the branch
> CAM-4450-timer-event-listener
> <https://github.com/camunda/camunda-bpm-platform/tree/CAM-4450-timer-event-listener>.
>
>
> So, the next steps would be:
> 1) A JobDefinition must be created, when deploying a case definition
> containing a timer event listener
> 2) Implement the timer event listener job that the event occurred is
> triggered when the timer job is executed.
> 3) Implement the start trigger
> 4) Implement repeating timer intervals
>
> I would appreciate it, if you could create for each point a pull request.
>
> If you need some assistance, do not hesitate to ask me.
>
> Thanks for your effort!
>
> Cheers,
> Roman
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/camunda/camunda-bpm-platform/pull/222#issuecomment-208213738>
>

romansmirnov commented:
Hi Subhro,

Please create a new pull request with your current state. It makes it easier to approach you.

Thanks.

Cheers,
Roman

subhrajyotim commented:
Yes, Roman, Will do.

Thanks,
Subhro.

On Wed, Apr 20, 2016 at 2:11 PM, Roman Smirnov <notifications@github.com>
wrote:

> Hi Subhro,
>
> Please create a new pull request with your current state. It makes it
> easier to approach you.
>
> Thanks.
>
> Cheers,
> Roman
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/camunda/camunda-bpm-platform/pull/222#issuecomment-212331310>
>
